File: jupyter2/problems/oop_basics/class/go/class.go
package main
import (
	"fmt"
	"math"
)

/** begin my answer */

/** end my answer */
func veryClose(x, y float64) bool {
	return math.Abs(x - y) < 1e-6
}

func main() {
	e0 := mkFreeEllipse(16., 5.)
	e1 := mkFreeEllipse( 6., 3.)
	if ! veryClose(e0.area(), 251.327412) { panic("wrong") }
	if ! veryClose(e1.area(), 56.548668)  { panic("wrong") }
	if !(smaller(e0, e1) == 1) { panic("wrong") }
	fmt.Println("OK")
}

File: jupyter2/problems/oop_basics/class/rs/class.rs
/** begin my answer */

/** end my answer */

fn very_close(x: f64, y: f64) -> bool {
	(x - y).abs() < 1e-6
}

fn main() {
    let e0 = mk_free_ellipse(16., 5.);
    let e1 = mk_free_ellipse( 6., 3.);
    assert!(very_close(e0.area(), 251.327412));
    assert!(very_close(e1.area(), 56.548668));
    assert!(smaller(&e0, &e1) == 1);
    println!("OK")
}
